src/main/java/nl/tudelft/semgroup4/GameState.java

```diff
@@ -110,34 +105,44 @@ public void update(GameContainer container, StateBasedGame mainApp, int delta)
             throws SlickException {
         // checks if the escape key is pressed
         if (input.isKeyPressed(Input.KEY_ESCAPE)) {
-            // If the game is paused and the pause screen is openend, or if the
-            // game isn't paused, this code is executed. This prevents the user from
-            // being able to unpause the game while the countdown is running (because then
-            // the game is paused without the pause screen being open)
-            if ((currentGame.isPaused() && pauseScreenOpened)
-                    || !(currentGame.isPaused() || pauseScreenOpened)) {
-                Game.LOGGER.log(LogSeverity.DEBUG, "Game", "Player "
-                        + (currentGame.isPaused() ? "resumed" : "paused") + " the game");
-                input.disableKeyRepeat();
-                currentGame.setPaused(!currentGame.isPaused());
-                pauseScreenOpened = !pauseScreenOpened;
-            }
+            pauseScreenOpened = gamestateController.togglePauseMenu(pauseScreenOpened, input);
         }
 
-        if (!currentGame.isPaused()) {
-            currentGame.update(delta);
-        } else {
-            currentGame.getCountdown().update();
-        }
+        gamestateController.updateGame(delta);
         dashboard.update(delta);
     }
+    
+    @Override
+    public int getID() {
+        return States.GameState;
+    }
 
     protected Game getGame() {
         return currentGame;
     }
-
+    
+    protected void setGameStateController(GameStateController controller) {
+        gamestateController = controller;
+    }
+    
+    protected void setDashboard(Dashboard dashboard) {
+        this.dashboard = dashboard;
+    }
+    
+    protected PauseScreen getPausescreen() {
+        return pauseScreen;
+    }
+    
+    protected boolean getPauseScreenOpened() {
```

Comment:
PMD nags that this should be called <code>isPauseScreenOpenened</code>.

< naming